libres/lib/analysis/ies/ies.cpp

```diff
@@ -138,27 +149,21 @@ Eigen::MatrixXd ies::makeX(const Eigen::MatrixXd &A, const Eigen::MatrixXd &Y0,
      * ies_inversion=IES_INVERSION_EXACT(0)            -> exact inversion from (b) with exact R=I
      * ies_inversion=IES_INVERSION_SUBSPACE_EXACT_R(1) -> subspace inversion from (a) with exact R
      * ies_inversion=IES_INVERSION_SUBSPACE_EE_R(2)    -> subspace inversion from (a) with R=EE
-     * ies_inversion=IES_INVERSION_SUBSPACE_RE(3)      -> subspace inversion from (a) with R represented by E
+     * ies_inversion=IES_INVERSION_SUBSPACE_RE(3)      -> subspace inversion from (a) with R represented by E * E^T
      */
 
     if (ies_inversion != ies::config::IES_INVERSION_EXACT) {
         ies::linalg_subspace_inversion(W0, ies_inversion, E, R, S, H,
                                        truncation, ies_steplength);
     } else if (ies_inversion == ies::config::IES_INVERSION_EXACT) {
-        ies::linalg_exact_inversion(W0, ies_inversion, S, H, ies_steplength);
+        ies::linalg_exact_inversion(W0, S, H, ies_steplength);
     }
 
-    /*
-     * CONSTRUCT TRANFORM MATRIX X FOR CURRENT ITERATION (Line 10)
-     *   X= I + W/sqrt(N-1)
-     */
+    /* Line 9 of Algorithm 1 */
     Eigen::MatrixXd X = W0;
     X /= sqrt(ens_size - 1.0);
     X.diagonal().array() += 1;
 
-    /* COMPUTE ||W0 - W|| AND EVALUATE COST FUNCTION FOR PREVIOUS ITERATE (Line 12)*/
-    Eigen::MatrixXd DW = W0 - W;
```

Comment:
Was this never used?

Reply:
Looks like it was used when we wrote to files for use with Tecplot.
Nobody has complained yet, so I think it's ok to delete.

```cpp
matrix_type *DW = matrix_alloc(ens_size, ens_size);
matrix_sub(DW, W0, W);
teclog(W, D, DW, "iesteclog.dat", ens_size, iteration_nr, rcond, nrsing,
       nrobs_inp);
teccost(W, D, "costf.dat", ens_size, iteration_nr);
```